### PR TITLE
Update contributing guide to be more friendly

### LINKED ISSUE #819: Building documentation using nox
### Is this issue already tracked somewhere, or is this a new report?

- [X] I've reviewed existing issues and couldn't find a duplicate for this problem.

### Current Behavior

We haven't configured `nox -s build_docs -- --serve` 

### Expected Behavior

`nox -s build_docs -- --serve` should work 

### Steps To Reproduce

1. clone the repository 
2. run `nox -s build_docs -- --serve`

### Environment

```markdown
- OS:
- Python:
```

### Additional Context

We'll need to add a section in the docs to clarify that integration tests and documentation will require that we export valid EDL credentials with certain EULAs approved.

### PR DESCRIPTION
**Pull request by jhkennedy**
`earthaccess` documentation currently states that creating an issue is **required** before submitting a PR, which I think is an unnecessary barrier to many possible contributions. For example, it makes little sense to require an issue to fix a documentation spelling error. More discussion in slack here: https://openscapes.slack.com/archives/C05TMK269HA/p1726880090555649

I've:
* removed the " Ensure an issue exists representing the problem being solved in this PR." checklist item from our PR template
* added a "thank you" as the first thing potential contributors see when coming the contributing docs
* expanded the discussion of how to contribute
* reworked the different PR paths to be tabs instead of sections
* fixed a few typos
* updated `releasing.md` to exclusively use MkDocs style admonitions so they render well in ReadTheDocs (it had a mix of GitHub style and MkDocs style, and the GitHub style ones don't render well in our docs website)
* moved some of the documentation details to `development.md` to improve the flow
* fixes #819 by adding a nox session for docs. Note:
   * `mkdocs build` does not have a `--serve` option as suggested in the issue so we need to use `mkdocs serve`
   * the `--dirtyreload` option to `mkdocs serve` has been renamed `--dirty`
   * I've removed `scripts/docs-live.sh` as it's been replaced by nox
   * I've also removed `scripts/build-docs.sh` as it appears to be unused and undocumented, but I _did not_ add a nox session to build the docs since ReadTheDocs does that for us. It would be easy to add in however.
 * updated all documented usages of `pip` to be invoked in the preferred method `python -m pip`, except the discussion of `pipx` b/c there you actually don't want your environment python to be invoked and instead use your system one (in reality this should be discussed more and I think this difference makes it maybe not suitable for the quickstart, but I'm leaving it for now).

<details><summary>Pull Request (PR) draft checklist - click to expand</summary>

- [x] Please review our
      [contributing documentation](https://earthaccess.readthedocs.io/en/latest/contributing/)
      before getting started.
- [x] ~~Ensure an issue exists representing the problem being solved in this PR.~~ <- no.
- [x] Populate a descriptive title. For example, instead of "Updated README.md", use a
      title such as "Add testing details to the contributor section of the README".
      Example PRs: [#763](https://github.com/nsidc/earthaccess/pull/763)
- [x] Populate the body of the pull request with:
    - A clear description of the change you are proposing.
    - Links to any issues resolved by this PR with text in the PR description, for
      example `closes #1`. See
      [GitHub docs - Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
- [ ] Update `CHANGELOG.md` with details about your change in a section titled
      `## Unreleased`. If such a section does not exist, please create one. Follow
      [Common Changelog](https://common-changelog.org/) for your additions.
      Example PRs: [#763](https://github.com/nsidc/earthaccess/pull/763)
- [x] Update the documentation and/or the `README.md` with details of changes to the
      earthaccess interface, if any. Consider new environment variables, function names,
      decorators, etc.

Click the "Ready for review" button at the bottom of the "Conversation" tab in GitHub
once these requirements are fulfilled. Don't worry if you see any test failures in
GitHub at this point!

</details>

<details><summary>Pull Request (PR) merge checklist - click to expand</summary>

Please do your best to complete these requirements! If you need help with any of these
requirements, you can ping the `@nsidc/earthaccess-support` team in a comment and we
will help you out!

- [ ] Add unit tests for any new features.
- [ ] Apply formatting and linting autofixes. You can add a GitHub comment in this Pull
      Request containing "pre-commit.ci autofix" to automate this.
- [ ] Ensure all automated PR checks (seen at the bottom of the "conversation" tab) pass.
- [ ] Get at least one approving review.

</details>

----
📚 Documentation preview 📚: https://earthaccess--837.org.readthedocs.build/en/837/

### CODE REVIEW

**Comment by jhkennedy** [comment]
I expect we'll want to iterate quite a bit on language, but this should get the discussion going!

**Comment by jhkennedy** [comment]
Oh, and for #819, I have note done this:
> We'll need to add a section in the docs to clarify that integration tests and documentation will require that we export valid EDL credentials with certain EULAs approved.

Nor have I updated the `CHANGELOG.md` yet; figured we should settle on the PR and circle back to that.

File: docs/contributing/development.md [hunk 1]
```diff
@@ -24,8 +27,8 @@ specific jobs:
 ```console
 $ nox -s typecheck              # Typecheck only
 $ nox -s tests                  # Python tests
-$ nox -s build_docs -- --serve  # Build and serve the docs
-$ nox -s build_pkg              # Make an SDist and wheel
+$ nox -s serve_docs             # build and serve the docs
```
Comment (jhkennedy) [inline docs/contributing/development.md:30]:
```suggestion
$ nox -s serve_docs             # Build and serve the docs
```

File: docs/contributing/development.md [hunk 2]
```diff
@@ -2,8 +2,11 @@
 
 ## Getting the code
 
-1. Fork [nsidc/earthaccess](https://github.com/nsidc/earthaccess)
-1. Clone your fork (`git clone git@github.com:{my-username}/earthaccess`)
+1. Fork [nsidc/earthaccess](https://github.com/nsidc/earthaccess/fork)
```
Comment (mfisher87) [inline docs/contributing/development.md:5]:
```suggestion
1. [Fork nsidc/earthaccess](https://github.com/nsidc/earthaccess/fork)
```

File: docs/contributing/development.md [hunk 3]
```diff
@@ -2,8 +2,11 @@
...
+1. Clone your fork
```
Comment (mfisher87) [inline docs/contributing/development.md:6]:
```suggestion
1. Clone your fork:
```

File: docs/contributing/development.md [hunk 4]
```diff
@@ -36,33 +39,86 @@ environment for each run.
 While `nox` is the fastest way to get started, you will likely need a full
 development environment for making code contributions, for example to test in a
 REPL, or to resolve references in your favorite IDE.  This development
-environment also includes `nox`.
+environment also includes `nox`. You can create it with `venv`, `conda`, or `mamba`.
 
-Create and activate a virtual environment with `venv`, which comes by default
-with Python, in the `.venv` directory:
+=== "`venv`"
```
Comment (mfisher87) [inline docs/contributing/development.md:44]: Is this a special type of admonition? Turns out it's hard to search for this.

File: docs/contributing/development.md [hunk 5]
```diff
@@ -36,33 +39,86 @@ environment for each run.
...
+## Usage of Pre-Commit
+
+To maintain code quality, we use pre-commit for automated checks before committing changes. We recommend you install
+and configure pre-commit so you can format and lint as you go instead of waiting for CI/CD check failures and
+having to make significant changes to get the checks to pass.
+
+To set up pre-commit, follow these steps:
+
+- `python -m pip install pre-commit` ([official installation docs](https://pre-commit.com/#install))
+- `pre-commit install` to enable it to run automatically
```
Comment (mfisher87) [inline docs/contributing/development.md:93]:
```suggestion
- `pre-commit install` to enable it to run automatically for each commit
```

File: docs/contributing/development.md [hunk 6]
```diff
@@ -36,33 +39,86 @@ environment for each run.
...
+- `pre-commit run -a`  to run it manually
+
+We have included type stubs for the untyped `python-cmr` library, which we intend to eventually upstream.
```
Comment (mfisher87) [inline docs/contributing/development.md:96]:
```suggestion
## Type stubs

We have included type stubs for the untyped `python-cmr` library, which we intend to eventually upstream.
```

File: docs/contributing/index.md [hunk 7]
```diff
@@ -1,99 +1,88 @@
 # Contributing
 
-When contributing to this repository, please first discuss the change you wish to make
-with the community and maintainers via
-[a GitHub issue](https://github.com/nsidc/earthaccess/issues),
-[a GitHub Discussion](https://github.com/nsidc/earthaccess/discussions),
-or [any other method](our-meet-ups.md).
+Thank you for your interest in contributing to _earthaccess_! We're excited for your contribution,
+whether you're finding bugs, adding new features, fixing anything broken, or improving documentation.
```
Comment (mfisher87) [inline docs/contributing/index.md:4]: :100:

File: docs/contributing/index.md [hunk 8]
```diff
@@ -1,99 +1,88 @@
...
-When you feel comfortable, fork this repository and set up your development environment.
-Don't forget to run our suite of checks (see [development environment
-documentation](./development.md) for details)!
+- Open [a GitHub issue](https://github.com/nsidc/earthaccess/issues) to report a bug, unexpected behavior, a stumbling block in our documentation, or any
+  other problem.
+- Open [a Q&A GitHub Discussion](https://github.com/nsidc/earthaccess/discussions/categories/new?category=q-a) to ask
+  questions, share insights, or connect with others about Earth data access (broadly)
+- Open [an Ideas GitHub Discussion](https://github.com/nsidc/earthaccess/discussions/new?category=ideas) to suggest
+  new features or improvements, and find collaborators and coordinate the work
+- Join [a "hack day"](our-meet-ups.md) to meat other users/developers/maintainers and get help on, or give help to,
```
Comment (mfisher87) [inline docs/contributing/index.md:20]: :meat_on_bone: :)

```suggestion
- Join [a "hack day"](our-meet-ups.md) to meet other users/developers/maintainers and get help on, or give help to,
```

File: docs/contributing/index.md [hunk 9]
```diff
@@ -1,99 +1,88 @@
...
+  anything related to Earth data access
 
-## Sharing your work with us on GitHub
+You can also directly [open a pull request](#steps-to-a-pull-request) after you've reviewed this whole contributing guide.
 
-From here, you might want to fix and issue or bug, or add a new feature.  Jump to the
-relevant section to proceed.
+If you're not sure what to do, _don't worry_, and just pick whichever suites you best.
```
Comment (mfisher87) [inline docs/contributing/index.md:25]:
```suggestion
If you're not sure what to do, _don't worry_, and just pick whichever suits you best. The community will help you out!
```

File: docs/contributing/index.md [hunk 10]
```diff
@@ -1,99 +1,88 @@
...
-- Run the unit tests successfully in your branch
-- Write the code to implement your new feature in a backwards compatible manner
-- Create at least one test that exercises your feature and run the test suite as you go
+From here, you might want to fix and issue or bug, or add a new feature. Jump to the
+relevant tab to proceed.
 
-Once you've completed these steps, you are ready to submit your contribution.
+!!! tip
 
+    The small the pull request, the easier it is to  review and test, and the more likely it is to be successful.
```
Comment (mfisher87) [inline docs/contributing/index.md:40]:
```suggestion
    The smaller the pull request, the easier it is to review and test, and the more likely it is to be successful. For more details, check out this [helpful YouTube video by the author of pre-commit](https://www.youtube.com/watch?v=Gu6XrmfwivI.
```

File: docs/contributing/index.md [hunk 11]
```diff
@@ -1,99 +1,88 @@
...
+    - Create a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example), a.k.a. a Minimal Complete Verifiable Example (MCVE),
+      a Minimal Working Example (MWE), an SSCCE (Short, Self Contained, Complete Example), or a "reprex",
+      and include it in [a GitHub issue](https://github.com/nsidc/earthaccess/issues) (recommended) or in your pull request.
+    - Create a branch to resolve your issue
+    - Run the unit tests successfully in your branch
+    - Create one or more new tests to demonstrate the bug and observe them fail
+    - Update the relevant code to fix the issue
+    - Successfully run your new unit tests
+    - Lint and format your code with [Pre-Commit](development.md#usage-of-pre-commit)
+    - Described your changes in the `CHANGELOG.md`
```
Comment (mfisher87) [inline docs/contributing/index.md:57]:
```suggestion
    - Describe your changes in the `CHANGELOG.md`
```

File: docs/contributing/index.md [hunk 12]
```diff
@@ -1,99 +1,88 @@
...
 
-#### Documentation Style
+=== "Fixing an Issue or Bug"
 
-To ensure that our code is well-documented and easy to understand, we use [Google-style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) to document all functions, classes, and methods in this library.
+    - Create a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example), a.k.a. a Minimal Complete Verifiable Example (MCVE),
+      a Minimal Working Example (MWE), an SSCCE (Short, Self Contained, Complete Example), or a "reprex",
+      and include it in [a GitHub issue](https://github.com/nsidc/earthaccess/issues) (recommended) or in your pull request.
+    - Create a branch to resolve your issue
+    - Run the unit tests successfully in your branch
```
Comment (mfisher87) [inline docs/contributing/index.md:52]:
```suggestion
    - Run the existing unit tests successfully in your branch
```

File: docs/contributing/index.md [hunk 13]
```diff
@@ -1,99 +1,88 @@
...
+    - Successfully run your new unit tests
+    - Lint and format your code with [Pre-Commit](development.md#usage-of-pre-commit)
+    - Described your changes in the `CHANGELOG.md`
 
-#### Locally rendering the documentation
+=== "Contributing a New Feature"
 
-To work on documentation locally, we provide a script that will automatically re-render the docs when you make changes:
+    - We recommend you create [an Ideas GitHub Discussion](https://github.com/nsidc/earthaccess/discussions/new?category=ideas)
+      describing the feature's scope and it's fit for this package with the team
```
Comment (mfisher87) [inline docs/contributing/index.md:62]:
```suggestion
      describing the feature's scope and its fit for this package with the team.
```

File: docs/contributing/index.md [hunk 14]
```diff
@@ -1,99 +1,88 @@
...
+    - Create a branch for your new feature in your fork
+    - Run the unit tests successfully in your branch
+    - Write the code to implement your new feature in a backwards compatible manner
```
Comment (mfisher87) [inline docs/contributing/index.md:65]:
```suggestion
    - Write the code to implement your new feature in a backwards compatible manner. If breaking changes are necessary, discuss your strategy with the team first.
```

File: docs/contributing/index.md [hunk 15]
```diff
@@ -1,99 +1,88 @@
...
+    - Create at least one test that exercises your feature and run the test suite
+    - Lint and format your code with [Pre-Commit](development.md#usage-of-pre-commit)
```
Comment (mfisher87) [inline docs/contributing/index.md:67]:
```suggestion
    - Lint and format your code with [pre-Commit](development.md#usage-of-pre-commit)
```

**Review by mfisher87 — APPROVED** [review]
Awesome work. I have some nitpicks! I don't feel the need to review again once they are addressed, but am happy to!

File: docs/contributing/index.md (hunk 10 above)
Comment (jhkennedy) [reply docs/contributing/index.md:40]: Nice reference!

File: docs/contributing/development.md [hunk 16]
```diff
@@ -36,33 +39,88 @@ environment for each run.
...
+    This will update (or create if missing) the `earthaccess` environment and active it. The `earthaccess` package will
+    be installed into the environment in editable mode with the optional development dependencies.
+
 ## Managing Dependencies
 
 If you need to add a new dependency, edit `pyproject.toml` and insert the
 dependency in the correct location (either in the `dependencies` array or under
 `[project.optional-dependencies]`).
+
+## Usage of Pre-Commit
```
Comment (jhkennedy) [inline docs/contributing/development.md:84]:
```suggestion
## Usage of pre-commit
```

Matching pre-commit's preferred styling

File: docs/contributing/index.md [hunk 17]
```diff
@@ -1,99 +1,88 @@
...
-To ensure that our code is well-documented and easy to understand, we use [Google-style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) to document all functions, classes, and methods in this library.
+    - Create a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example), a.k.a. a Minimal Complete Verifiable Example (MCVE),
+      a Minimal Working Example (MWE), an SSCCE (Short, Self Contained, Complete Example), or a "reprex",
+      and include it in [a GitHub issue](https://github.com/nsidc/earthaccess/issues) (recommended) or in your pull request.
+    - Create a branch to resolve your issue
+    - Run the existing unit tests successfully in your branch
+    - Create one or more new tests to demonstrate the bug and observe them fail
+    - Update the relevant code to fix the issue
+    - Successfully run your new unit tests
+    - Lint and format your code with [Pre-Commit](development.md#usage-of-pre-commit)
```
Comment (jhkennedy) [inline docs/contributing/index.md:56]:
```suggestion
    - Lint and format your code with [pre-commit](development.md#usage-of-pre-commit)
```

File: mkdocs.yml [hunk 18]
```diff
@@ -120,6 +120,8 @@ markdown_extensions:
   - pymdownx.inlinehilite
   - pymdownx.snippets
   - pymdownx.superfences
+  - pymdownx.tabbed:
+      alternate_style: true
```
[the remark this answers is not in the document]
Comment (jhkennedy) [reply mkdocs.yml:124]: ¯\_(ツ)_/¯

**Review by mfisher87 — APPROVED** [review]
:rocket:

File: docs/contributing/index.md [hunk 19]
```diff
@@ -1,99 +1,88 @@
...
+- You've included an explanation of what you did and why in the pull request
+- All unit tests run successfully in your branch
+- Your code is linted and formatted correctly
+- The documentation and `CHANGELOG.md` has been updated appropriately
 
-Fork the repository using the "Fork" button on the [repository
-homepage](https://github.com/nsidc/earthaccess), create a branch with your changes in the fork, then open
-a draft pull request from your fork. Starting a pull request provides additional instructions and requirements, and
-there is no harm in starting a draft pull request while still developing.
+Then you can mark the pull request as ready for review! :tada:
```
Comment (jhkennedy) [inline docs/contributing/index.md:88]:
```suggestion
Then you can mark the pull request as ready for review! 🎉
```
MkDocs apparently doesn't do `:tada:` -> 🎉 (there's probably an extension for that)

**Pull request merged**